agent/action/smart_action.py

```
@@ -0,0 +1,396 @@
+            utils.mfaalog.error(f"{tag} ❌ on_changed 与 on_unchanged 不能同时为 loop（这是死循环）")
+            return None
+        # loop_exhausted 是 loop 的出口，再填 loop 就没有出口了。想无限转请调大 loop_limit。
+        if loop_exhausted not in ("next", "error"):
+            utils.mfaalog.error(f"{tag} ❌ loop_exhausted 只接受 next/error，实际: {loop_exhausted!r}")
+            return None
+
+        try:
+            threshold = float(params.get("threshold", 3.0))
+            settle_delay = int(params.get("settle_delay", 500))
+            unchanged_streak = int(params.get("unchanged_streak", 1))
+            # 默认 20 而不是更大的数：它同时也是 on_unchanged:"loop" 的重试上限，而那一侧
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)